### PR TITLE
[WIP] Rename from API to Http-Client

### PR DESCRIPTION
**Pull request by AliabbasMerchant**
I'll keep on updating `renaming_status.txt`.  
The file will be deleted before the final merge.   

(The module may be unusable while the process of renaming is going on.)

### CODE REVIEW

**Comment by alexander255** [comment]
Thanks for working on this again!
One comment after looking over the first version: Do you really want to rename the root package from `ipfsapi` to `ipfs_http_client`? Why not use `ipfshttpclient` instead, as it is standard python convention?

**Comment by AliabbasMerchant** [comment]
@daviddias @alexander255  
I have made all changes from my side.  
Please have a look at `renaming_status.txt`  
I have marked a few things as doubtful

**Comment by alexander255** [comment]
Looks mostly good. Please take a look at those CI failures (aside from codestyle). Also please don't create the `-ipfshttpclient.*` files – just leave them at their original name. Once I merge this I'll take care to split off the previous version into another branch and start working on that compat-shim I talked about.

**Comment by AliabbasMerchant** [comment]
1. I'll look into the CI failure.  
2. I havent renamed any of the files in `ipfshttpclient`. Just changed the folder name from `ipfsapi` to `ipfshttpclient`, and changed the contents of the files.

**Comment by alexander255** [comment]
> I havent renamed any of the files in ipfshttpclient. Just changed the folder name from ipfsapi to ipfshttpclient, and changed the contents of the files.

Then please delete the original files (`git rm`), commit that and [squash your commits](https://stackoverflow.com/a/5201642/277882). GIT will then detect the files as “renamed with changes” and `git blame` will continue to work. As I said please also rename the `-ipfshttpapi.*` files back to their original names.

Thanks for taking the time to work on this and all the work you've already submitted thus far!

**Comment by AliabbasMerchant** [comment]
@alexander255 I have renamed the 3 `-ipfshttpclient.*` files back to their original name and have made changes in them, under the original name itself.  
Also, I have squashed all the commits.  
With regards to `docs/api_ref.md` -> `docs/http_client_ref.md`, no matter what I do, git is considering it as a delete-new_file and not a rename.

File: renaming_status.txt [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+External Changes to be made:
+(See the section under "FOR README.md")
+
+
+THESE MAY REFER TO THE ACTUAL API
+ipfshttpclient/client.py          All occurrences
+ipfshttpclient/encoding.py        Ln 392
+ipfshttpclient/http.py            Ln 2,4,134
+ipfshttpclient/version.py         Ln 7
+test/run-tests.py                 Ln 68,102
```
Comment (alexander255) [inline renaming_status.txt]: Reviewed all these and all, except for the ones in `client.py`, indeed refer to the IPFS HTTP API and should be changed. Current state looks good.

File: renaming_status.txt [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+
+FOR README.md:
+* I have made changes in the original file itself
+* External changes to be done by the project maintainers (I have already made the changes in the file, on the respective lines):
+    https://img.shields.io/pypi/v/ipfsapi.svg?style=flat-square -> https://img.shields.io/pypi/v/ipfshttpclient.svg?style=flat-square   line 6
```
Comment (alexander255) [inline renaming_status.txt]: Will happen automatically when we push the new PyPI package.

File: renaming_status.txt [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+    https://pypi.python.org/pypi/ipfsapi -> https://pypi.python.org/pypi/ipfshttpclient                                                 line 6
```
Comment (alexander255) [inline renaming_status.txt]: Will need to be manually just after this branch is merged. Needs transitional code for old package in place.

File: renaming_status.txt [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    https://travis-ci.org/ipfs/py-ipfs-api.svg?branch=master -> https://travis-ci.org/ipfs/py-ipfs-http-client.svg?branch=master        line 7
+    https://travis-ci.org/ipfs/py-ipfs-api -> https://travis-ci.org/ipfs/py-ipfs-http-client                                            line 7
```
Comment (alexander255) [inline renaming_status.txt]: Will happen automatically once the source repo is renamed.

File: renaming_status.txt [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+    https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfsapi/ ->
+                    https://ipfs.io/ipns/QmZ86ow1byeyhNRJEatWxGPJKcnQKG7s51MtbHdxxUddTH/Software/Python/ipfshttpclient/                 line 11,118
```
Comment (alexander255) [inline renaming_status.txt]: Needs docs to be regenerated and pushed by me.

File: renaming_status.txt [hunk 6]
```diff
@@ -0,0 +1,81 @@
...
+    The PIP package and Python module: ipfshttpclient
```
Comment (alexander255) [inline renaming_status.txt]: Needs transitional module for old name in-place.

File: renaming_status.txt [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+    The Github repo: py-ipfs-api -> py-ipfs-http-client
```
Comment (alexander255) [inline renaming_status.txt]: Needs help from org maintainers iirc.

File: renaming_status.txt [hunk 8]
```diff
@@ -0,0 +1,81 @@
...
+    PROJECT RENAME:        Python IPFS API -> Python IPFS HTTP Client
```
Comment (alexander255) [inline renaming_status.txt]: Happens automatically when this is merged.

File: test/unit/test_http.py [hunk 9]
```diff
@@ -219,18 +219,19 @@ def test_session(self):
                 assert res == b'okay'
             assert self.client._session is None
 
+
 def test_stream_close(mocker):
-    client = ipfsapi.http.HTTPClient("localhost", 5001, "api/v0")
-    mocker.patch("ipfsapi.http._notify_stream_iter_closed")
+    client = ipfshttpclient.http.HTTPClient("localhost", 5001, "http_client/v0")
```
Comment (alexander255) [inline test/unit/test_http.py]: This is refering to the Go-IPFS API endpoint. You cannot change this.

File: test/unit/test_http.py [hunk 10]
```diff
@@ -139,10 +139,10 @@ class TestHttp(unittest.TestCase):
     """
     def setUp(self):
         """Creates an instance of HTTPClient to test against."""
-        self.client = ipfsapi.http.HTTPClient(
+        self.client = ipfshttpclient.http.HTTPClient(
             'localhost',
             5001,
-            'api/v0')
+            'http_client/v0')
```
Comment (alexander255) [inline test/unit/test_http.py]: This is refering to the Go-IPFS API endpoint. You cannot change this.

**Comment by AliabbasMerchant** [comment]
Ok. I'll do the required changes. Just give me a few days time. I am currently not at home and don't have my laptop. :)

**Comment by AliabbasMerchant** [comment]
Finally! No conflicts.  
The file `api_ref.md` was successfully renamed to `http_client_ref.md`  
Also, I have changed `http_client/v0` to `api/v0`

File: test/unit/test_http.py [hunk 11]
```diff
@@ -170,45 +170,45 @@ def test_stream(self):
...
+        with HTTMock(http_client_cat):
             res = self.client.request('/cat')
             assert res == b'{"Message": "do not parse"}'
 
     def test_default_decoder(self):
         """Tests that the default encoding is set to json."""
-        with HTTMock(api_okay):
-            res = self.client.request('/apiokay')
+        with HTTMock(http_client_okay):
+            res = self.client.request('/http_client_fail')
```
Comment (alexander255) [inline test/unit/test_http.py]: This should be `/http_client_okay`

File: test/unit/test_http.py [hunk 12]
```diff
@@ -170,45 +170,45 @@ def test_stream(self):
...
             assert res == b'{"Message": "okay"}'
 
     def test_explicit_decoder(self):
         """Tests that an explicit decoder is handled correctly."""
-        with HTTMock(api_okay):
-            res = self.client.request('/apiokay',
+        with HTTMock(http_client_okay):
+            res = self.client.request('/http_client_fail',
```
Comment (alexander255) [inline test/unit/test_http.py]: Same here

**Pull request merged**